docs/plugin-transform-react-jsx-development.md

```diff
@@ -0,0 +1,122 @@
+---
+id: babel-plugin-transform-react-jsx-development
+title: "@babel/plugin-transform-react-jsx-development"
+sidebar_label: transform-react-jsx-development
+---
+
+:::info
+This plugin is included in `@babel/preset-react`
+:::
+
+This plugin is a developer version of [`@babel/plugin-transform-react-jsx`](./plugin-transform-react-jsx.md). It is designed to provide enhanced validation error messages and precise code location information for debugging React apps. Note that this plugin is intended to be used in a development environment, as it generates significantly more outputs than the production build.
+
+If you are using [`@babel/preset-react`](./preset-react.md), it will be automatically enabled by the [`development`](./preset-react.md#development) option so you don't have to install it.
+
+## Example
+
+**In**
+
+```js title="input.jsx"
+const profile = (
+  <div>
+    <img src="avatar.png" className="profile" />
+    <h3>{[user.firstName, user.lastName].join(" ")}</h3>
+  </div>
+);
+```
+
+**Out**
+
+```js title="output.js"
+import { jsxDEV as _jsxDEV } from "react/jsx-dev-runtime";
+
+const _jsxFileName = "input.jsx";
+const profile = _jsxDEV("div", {
+  children: [
+    _jsxDEV("img", {
+      src: "avatar.png",
+      className: "profile",
+    }, undefined, false, { fileName: _jsxFileName, lineNumber: 3, columnNumber: 5 }, this),
+    _jsxDEV("h3", {
+      children: [user.firstName, user.lastName].join(" "),
+    }, undefined, false, { fileName: _jsxFileName, lineNumber: 4, columnNumber: 5 }, this),
+  ]},
+  undefined, false, { fileName: _jsxFileName, lineNumber: 2, columnNumber: 3 }, this
+);
+```
```

Comment:
This example is taken from the `plugin-jsx` docs. I intend to provide only this one just to showcase the verbose output and the imports to `jsx-dev-runtime`. Developers can checkout the production JSX plugin pages for more examples.